conan/tools/cmake/toolchain/toolchain.py

```diff
@@ -184,6 +184,18 @@ def content(self):
         content = relativize_generated_file(content, self._conanfile, "${CMAKE_CURRENT_LIST_DIR}")
         return content
 
+    def _find_cmake_exe(self):
+        for req in self._conanfile.dependencies.direct_build.values():
+            if req.ref.name == "cmake":
```

Comment:
This assumes a package called `cmake` - which indeed is likely what it's going to be called, but I'm aways wary of blurring the separation between the client and the packages (which may or may not exist). However this is likely to be the package name and it is harmless otherwise :D

Question: what happens when the profile has a `platform_tool_requires` on cmake?

Reply:
> This assumes a package called `cmake` - which indeed is likely what it's going to be called, but I'm aways wary of blurring the separation between the client and the packages (which may or may not exist). However this is likely to be the package name and it is harmless otherwise :D

Yes, I agree, that was the same thing I was not 100% convinced about this.

> 
> Question: what happens when the profile has a `platform_tool_requires` on cmake?

Good point, I'll check if something weird happens in that case.

Reply:
The ``platform_tool_requires`` in theory shouldn't get to this point in the CMakeToolchain code, it is filtered before, so it will not be in ``direct_build`` dependencies?

Reply:
Yes, that's exactly what happens, I expanded the test with that to verify anyway